webview-ui/src/App.tsx

```diff
@@ -76,85 +88,122 @@ const App = () => {
     });
   });
 
-  if (state.state === "loading") {
+  useEffect(() => {
+    if (props.dataSourceURL && state.state === "loading") {
+      console.log("Loading data from", props.dataSourceURL);
+      fetch(props.dataSourceURL)
+        .then((response) => response.text())
+        .then((text) => {
+          loadData({ lines: text });
+        });
+    }
+  }, [props.dataSourceURL, loadData, state.state]);
 
-    return <div className="flex h-dvh items-center justify-center">
-      <Card className="">
-        <label htmlFor="data-input" className="block font-bold mb-2">Visualize from File(s)</label>
-        <input className="block font-sans rounded-md cursor-pointer focus:outline-none file:border-0 file:bg-primary file:text-white"
-          multiple id="data-input" type="file"
-          onChange={(e: any) => {
-            const files: FileList = e.target.files;
-            const reader = new FileReader();
-            let lines = "";
-            let i = 0;
-            reader.onload = (e: any) => {
-              lines = lines + "\n" + e.target.result;
-              i++;
-              loadData({ lines });
-              if (i < files.length) {
-                reader.readAsText(files[i]);
-              }
-            };
-            reader.readAsText(files[i]);
-          }} />
-      </Card>
-    </div>;
+  if (state.state === "loading") {
```

Comment:
Everything below this line is Prettier noise.

Side discussion: should we check in a `.prettierrc` to avoid future noise?